### PR TITLE
[INTERNAL] Clone resource when reading from Memory

### PR DESCRIPTION
**Pull request by flovogt**
Clone resources when reading them from the memory.

JIRA: CPOUI5FOUNDATION-567

Follow-Up of https://github.com/SAP/ui5-fs/pull/235.

### CODE REVIEW

File: lib/Resource.js [hunk 1]
```diff
@@ -323,6 +323,8 @@ class Resource {
 			options.createStream = this._createStream;
 		} else if (this._buffer) {
 			options.buffer = this._buffer;
+		} else if (this.__project) {
+			options.project = clone(this.__project);
```
[the remark this answers is not in the document]
Comment (flovogt) [reply lib/Resource.js]: No. Its a reference

File: test/lib/Resource.js [hunk 2]
```diff
@@ -316,6 +316,31 @@ test("Resource: clone resource with stream", async (t) => {
...
+	const resource = new Resource({
+		path: "my/path/to/resource",
+		project: myProject
+	});
+
+	const clonedResource = await resource.clone();
+	t.pass("Resource cloned");
+
+	const clonedResourceProject = await clonedResource.getProject();
+	t.deepEqual(clonedResourceProject, resourceOptions.project, "Cloned resource has correct project content");
```
Comment (RandomByte) [inline test/lib/Resource.js:339]: I would think that the t.is assert in the next line is sufficient. If the two variables hold the same instance, they are naturally deeply-equal too, are they not?

**Review by RandomByte — CHANGES_REQUESTED** [review]

File: test/lib/Resource.js (hunk 2 above)
Comment (flovogt) [reply test/lib/Resource.js:339]: Thats true... in one of the commits before i have cloned the project too, so the instance check was false but the deepequal check was true. Will remove this assert

**Comment by flovogt** [comment]
Closing this PR. Implementation is included in #448.

**Pull request closed without merging**